src/main/networking/cert-parse.ts

```diff
@@ -0,0 +1,249 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some code is derived from node-forge:
+
+New BSD License (3-clause)
+Copyright (c) 2010, Digital Bazaar, Inc.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Digital Bazaar, Inc. nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL DIGITAL BAZAAR BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+import forge, { asn1 } from 'node-forge';
+
+import Logging from '@/utils/logging';
+
+const console = Logging.background;
+
+const x509CertificateValidityValidator = {
+  name:        'Certificate',
+  tagClass:    asn1.Class.UNIVERSAL,
+  type:        asn1.Type.SEQUENCE,
+  constructed: true,
+  value:       [{
+    name:        'Certificate.TBSCertificate',
+    tagClass:    asn1.Class.UNIVERSAL,
+    type:        asn1.Type.SEQUENCE,
+    constructed: true,
+    value:       [{
+      name:        'Certificate.TBSCertificate.version',
+      tagClass:    asn1.Class.CONTEXT_SPECIFIC,
+      type:        0,
+      constructed: true,
+      optional:    true,
+      value:       [{
+        name:        'Certificate.TBSCertificate.version.integer',
+        tagClass:    asn1.Class.UNIVERSAL,
+        type:        asn1.Type.INTEGER,
+        constructed: false,
+      }]
```

Comment:
There's no trailing comma for the rest of the variable declaration; this is because this structure is unlikely to change in the future (it's a x.509 certificate).